File: smc/smc/report/level_wise_summary/level_wise_summary.py
# Copyright (c) 2013, Frappe Technologies and contributors
# For license information, please see license.txt

import frappe
from frappe.utils import cstr, cint, getdate, get_first_day, get_last_day, date_diff, add_days
from frappe import msgprint, _
from calendar import monthrange

def execute(filters=None):
	
	columns = get_columns(filters)
	conditions, filters = get_conditions(filters)
	data = get_data(conditions, filters)
	
	return columns, data

def get_data(conditions, filters):
	smc_fund = frappe.db.sql("""SELECT level,
	sum(case when gender = "Girls" then 1 else 0 end) as of_girls_schools,
	sum(case when gender = "Boys" then  1 else 0 end) as of_boys_schools,
	sum(case when gender = "Mixed" then  1 else 0 end) as of_mixed_school,
	round(sum(case when gender = "Girls" then 1 else 0 end)+sum(case when gender = "Boys" then  1 else 0 end)+sum(case when gender = "Mixed" then  1 else 0 end)) as total_school,
	sum(total_enrolment) as total_enrolment,
	sum(total_budget) as total_amount

	FROM `tabSMC Application Form` 
	WHERE docstatus =1 %s group by level_order"""% conditions, filters)
	return smc_fund

def get_conditions(filters):
	conditions=""
	if filters.get("year"):
		conditions = "  and year = %(year)s"
		user = frappe.session.user
		district = frappe.db.sql("select for_value from `tabUser Permission` where user='{0}' and allow='District' ".format(user), as_dict=True)
		if district:
			conditions += "  and district = '%s'"%(district[0].for_value)
	# if filters.get("year") and filters.get("district"):
	# 	conditions = "  and year = %(year)s and district = %(district)s"
	
	return conditions, filters

def get_columns(filters):
	columns = [
		_("Level") + "::160",
		_("# of Girls Schools") + "::180",
		_("# of Boys Schools") + "::180",
		_("# of Mixed Schools") + "::180",
		_("Total School") + "::180",
		_("# Total Enrolment") + "::180",
		 _("Total Amount") + ":Currency:180"
		]
		
	return columns
# @frappe.whitelist()
# def user_district(user):
# 	district = frappe.db.sql("select for_value from `tabUser Permission` where user=%s", (user), as_dict=True)[0]
# 	return district

File: smc/smc/report/level_wise_summary/level_wise_summary.js
// Copyright (c) 2016, Frappe Technologies and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Level Wise Summary"] = {
	"filters": [
		{
			"fieldname":"year",
			"label": __("Year"),
			"fieldtype": "Link",
			"options": "Year",
			"default":"2021-22",
			"reqd": 1
		},
		// {
		// 	"fieldname":"district",
		// 	"label": __("District"),
		// 	"fieldtype": "Link",
		// 	"options": "District",
		// 	// "hidden":1, 
		// 	"get_query": async function() {
		// 		let district_by_user;
		// 		var lsu = frappe.user_roles.includes('LSU');
		// 		var system_manager = frappe.user_roles.includes('System Manager');

		// 		if (lsu == true && system_manager == false) {
		// 			await frappe.call({
		// 				method: 'frappe.smc.report.level_wise_summary.level_wise_summary.user_district',
		// 				args: { user: frappe.session.user },
		// 				callback: function(r) {
		// 					district_by_user = r.message.for_value;
		// 					alert(r.message.for_value)
							
		// 				}
		// 			});
		// 		}	
		// 		return district_by_user
		// 	}
		// }
	]
};
